tuta-sdk/rust/sdk/src/crypto/kyber.rs

```diff
@@ -97,25 +104,57 @@ impl KyberPublicKey {
 	pub fn encapsulate(&self) -> KyberEncapsulation {
 		use pqcrypto_traits::kem::*;
 
-		let (shared_secret, ciphertext) = kyber1024_encapsulate(&self.public_key);
+		let (unbound_shared_secret, ciphertext) = mlkem1024_encapsulate(&self.public_key);
+
+		// turn ML-KEM into Kyber (except for some randomness hashing that we do not care about)
+		let shared_secret = bind_shared_secret_to_ciphertext(unbound_shared_secret, ciphertext);
 
 		KyberEncapsulation {
 			ciphertext: KyberCiphertext(ciphertext.as_bytes().try_into().unwrap()),
-			shared_secret: KyberSharedSecret(shared_secret.as_bytes().try_into().unwrap()),
+			shared_secret,
 		}
 	}
 }
 
-impl From<PQCryptoKyber1024PublicKey> for KyberPublicKey {
-	fn from(value: PQCryptoKyber1024PublicKey) -> Self {
+/// This is a redundant step to bind the derived shared secret to the ciphertext.
+/// It was part of the original round 3 Kyber submission specification and the reference implementation.
+/// It was removed from the NIST ML-KEM draft for efficiency because the re-encryption step in decapsulation prevents any attacks.
```

Comment:
I think a word is missing in this sentence :)

Reply:
I tried making it clearer. not sure which word is missing. is it ok now?